packages/react-dom/src/__tests__/ReactDeprecationWarnings-test.js

```diff
@@ -74,9 +86,7 @@ describe('ReactDeprecationWarnings', () => {
     );
   });
 
-  it('should not warn when owner and self are the same for string refs', () => {
-    ReactFeatureFlags.warnAboutStringRefs = false;
-
+  it('should warn when owner and self are the same for string refs', () => {
```

Comment:
As far as I can tell the added warning is not because of owner and self are the same but  simply because string refs are used.

It should still "not warn if owner and self are the same for string refs", right?

I think the test description is fine to keep while also not doing the `ReactFeatureFlags.warnAboutStringRefs = false;`.

`toErrorDev` should fail the test if we would suddenly add a warning "when owner and self are the same for string refs" because that would be a different warning than the one we're expecting now?